src/track/trackmetadatataglib.cpp

```diff
@@ -232,6 +232,71 @@ bool parseBpm(TrackMetadata* pTrackMetadata, QString sBpm) {
     return isBpmValid;
 }
 
+void parseRating(TrackMetadata *pTrackMetadata, QString sRating, FileType filetype = FileType::UNKNOWN) {
```

Comment:
The context (i.e. file/tag type) is known at compile time when this function is invoked. Please split it into 3 functions.